packages/admin-e2e-tests/src/pages/AnalyticsOverview.ts

```diff
@@ -93,6 +93,7 @@ export class AnalyticsOverview extends Analytics {
 	async addSection( sectionTitle: string ) {
 		await this.page.waitForSelector( "button[title='Add more sections']" );
 		await this.page.click( "button[title='Add more sections']" );
+		await waitForTimeout( 100 );
```

Comment:
@louwie17 Why do we need to wait **100ms** here? Maybe it would be great to add a comment?

Reply:
Good point @chihsuan , I added a comment, describing the addition:

// Delaying this, as there is a slight transition for the Add section button to show up.

Reply:
This doesn't make sense to me. Isn't the point of `waitForSelector` that it will wait until the element the selector specifies is available? Why does that not just work?

Having a timeout feels fragile.

Reply:
`waitForSelector` just checks if the element is present in the DOM, but not if CSS animations/transitions have finished. 
Prior to adding the timeout the tests would fail here sometimes, my guess being that the css transition hadn't finished yet and it tried to click on a wrong location.
I could either remove it and see how often it really happens or create a helper function that specifically waits for animations to finish, similar to this: https://stackoverflow.com/questions/51547994/wait-for-transition-to-end-in-puppeteer/57955251#57955251

Reply:
@mattsherman I end up mimicking something similar to [what Calypso did](https://github.com/Automattic/wp-calypso/pull/52089/files) (although they are using Selenium, but it solves the same problem), of creating a helper function that waits until a button stops moving, things seems to work pretty well now.

Reply:
@louwie17 That sounds like a good solution.